### PR TITLE
Add userbenchmark viewer

### PR DESCRIPTION
**Pull request by davidberard98**
Basic UI for viewing userbenchmark CSV results.

Usage: go to hud.pytorch.org/userbenchmark_view?url=[url]

where [url] is the path to the S3 csv, e.g.
```
https:%2F%2Fossci-metrics.s3.amazonaws.com%2Ftorchbench-pr-test%2Fpr84626%2Fresult.csv
```

These CSV results are generated on pytorch CI by the run_torchbench job, which runs if "RUN_TORCHBENCH: [userbenchmark name]" is added in the PR body. The userbenchmarks generate CSVs showing performance on a base revision and on the PR revision, for a number of metrics. The CSV is difficult to read on its own.

Screen Shot 2022-10-20 at 8 48 59 PM

### CODE REVIEW

File: torchci/pages/userbenchmark_view.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            as_array[i].push(base_value / pr_value - 1);
+        }
+        return as_array;
+    }
+
+    getHeader(row: any[]) {
+        const metric_header = row[0] as string;
+        const base_revision = row[1] + " (base)" as string;
+        const change_revision = row[2] + " (change)" as string;
+        const speedup_header = row[3] as string;
```
Comment (ZainRizvi) [inline torchci/pages/userbenchmark_view.tsx:47]: This seems like a good candidate for either a class or a [Named Tuple](https://www.geeksforgeeks.org/typing-namedtuple-improved-namedtuples/).  Getting the data indexes right can get tricky

File: torchci/pages/userbenchmark_view.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                const text = (speedupRaw * 100).toFixed(2) + '%';
+                // assign a color
+                let color = "rgb(255, 255, 255)";
+                if (speedupRaw >= 0) {
+                    const ratio = speedupRaw / maxSpeedup;
+                    const otherColor = Math.floor(255 * (1 - ratio)).toString();
+                    color = "rgb(" + otherColor + ", 255, " + otherColor + ")";
+                } else {
+                    const ratio = speedupRaw / minSpeedup;
+                    const otherColor = Math.floor(255 * (1 - ratio)).toString();
```
Comment (ZainRizvi) [inline torchci/pages/userbenchmark_view.tsx:72]: Might be worth extracting the 255 out to a const like `MAX_BRIGHTNESS`

File: torchci/pages/userbenchmark_view.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    render() {
+        return (
+            <div>
+                <h1> Userbenchmark results from <span>{this.state.url}</span> </h1>
+                <p>
+                    Userbenchmarks can be optionally run in the CI by adding
+                    &quot;RUN_TORCHBENCH: [userbenchmark]&quot; in the body of PRs
+                    in the pytorch repo, where [userbenchmark] should be replaced
```
Comment (ZainRizvi) [inline torchci/pages/userbenchmark_view.tsx:141]: I think you mean:

```suggestion
                    in the pytorch/pytorch repo, where [userbenchmark] should be replaced
```

**Review by ZainRizvi — APPROVED** [review]
Some readability tweaks, but looks good for the most part

File: torchci/pages/userbenchmark_view.tsx (hunk 1 above)
Comment (davidberard98) [reply torchci/pages/userbenchmark_view.tsx]: Added a class for this. But we still have to rely on indices when we first load the data from the csv.

**Review by ZainRizvi — APPROVED** [review]

**Review by ZainRizvi — APPROVED** [review]

**Pull request merged**